### PR TITLE
docker: added node to configuration and css lint setup

### PR DESCRIPTION
**Pull request by thelostone-mc**
##### Description

- reorganized folders
- updated docker-compose to install node
- created package.json + package-lock.json + stylelintrc + .editorconfig
- updated .gitignore

##### Checklist

- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/contributing/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)
- core
- doc

##### Refers/Fixes
Refs: #65

### CODE REVIEW

**Comment by thelostone-mc** [comment]
@mbeacom suggestions needed. 

added node to the docker image
__Flow:__  setup node via docker -> invoke `node-docker.bash` which will run commands to lint and stuff in the future.

@owocki  I've pushed the config + docs (all docs have a link in `README.md`) into subfolder to make it less cluttered.

PS: 
I've just added a basic `.stylelintrc` which can be invoked by running `npm run stylelint`.
Haven't re-linted the files yet (baby steps ) 

@Elaniobro , @ethikz thoughts on this ?

**Comment by Elaniobro** [comment]
Nice work @thelostone-mc! 

my only comment would be to add `eslint` to this mix too. Other than that, it looks good to me.

**Comment by ethikz** [comment]
Looks good @thelostone-mc!

**Comment by owocki** [comment]
@mbeacom this looks ok to me, but im not a docker guy.  🐳 

what do you think?

**Comment by mbeacom** [comment]
Overall, I'm good with this! Thanks for the addition @thelostone-mc.

The only thing I see possibly changing would be to install the NPM packages specifically to a docker volume versus to the locally mounted code volume.

Also, I rather move the requirements files to a `requirements` directory versus a `bin` directory for brevity.

**Comment by owocki** [comment]
> Also, I rather move the requirements files to a requirements directory versus a bin directory for brevity.

good point.

@thelostone-mc can you make it so?

**Comment by thelostone-mc** [comment]
@mbeacom verify verify verify :P

File: docker-compose.yml [hunk 1]
```diff
@@ -22,6 +22,11 @@ services:
       - db
     stdin_open: true
     tty: true
-
+  node:
+    image: node:9-alpine
+    working_dir: /node
+    volumes:
+      - .:/node
+    command: /bin/sh bin/docker-node.bash
```
Comment (mbeacom) [inline docker-compose.yml]: If you're going the `sh` route (I know, it's the default shell in Alpine), you might as well rename `docker-node.bash` to `docker-node.sh` and add the appropriate shebang to the top of the shell script:  `#!/bin/sh`.  Then you can replace `/bin/sh` with `bin/docker-node.bash`.

File: bin/docker-node.bash [hunk 2]
```diff
@@ -0,0 +1,2 @@
+npm install
```
Comment (mbeacom) [inline bin/docker-node.bash]: You can add the shebang here: `#!/bin/sh`

File: docker-compose.yml (hunk 1 above)
Comment (thelostone-mc) [reply docker-compose.yml]: Ah that makes sense! On it

**Comment by thelostone-mc** [comment]
@mbeacom changes done!
CI fails -> `Could not open requirements file: [Errno 2] No such file or directory: 'requirements-test.txt'`

thats strange though, it works fine locally : 
screen shot 2017-12-22 at 2 59 15 pm

**Comment by mbeacom** [comment]
@thelostone-mc Since you moved the requirements files, you'll have to update the travis.yml to install the requirements using the newly changed path at https://github.com/gitcoinco/web/blob/master/.travis.yml#L9

File: Dockerfile [hunk 3]
```diff
@@ -5,9 +5,9 @@ RUN mkdir /code && \
     apt-get install -y --no-install-recommends gcc libc6-dev libc-dev libssl-dev && \
     rm -rf /var/lib/apt/lists/*
 WORKDIR /code
-COPY requirements.txt /code/
-COPY requirements-test.txt /code/
+COPY base/requirements.txt /code/
```
Comment (mbeacom) [inline Dockerfile]: Sorry for the confusion... When we discussed the redundancy of `requirements/requirements-<type>.txt`, what I meant by moving to a `requirements` directory was to make the `requirements` directory and rename to something like: `requirements.txt` -> `base.txt`, `requirements-dev.txt` -> `dev.txt`, and `requirements-test.txt` -> `test.txt` and group them in `requirements`. I'm not locked into any structure, but `base/` might get confusing.

**Comment by thelostone-mc** [comment]
@mbeacom got it! Makes more sense

**Comment by thelostone-mc** [comment]
@owocki all done and dusted. 

Once this is merged, then `eslint` PR by @ethikz and I'll add scss loader later -> encouraging new css code to written to sass/scss ^_^

Then all that's left is the actual cleanup 😓

**Comment by owocki** [comment]
> Then all that's left is the actual cleanup 😓

😂

**Comment by thelostone-mc** [comment]
@ethikz We are good to go \m/

**Comment by ethikz** [comment]
Sorry everyone, been sick and unable to do anything :(

I am back on it now though, I have @thelostone-mc code pulled in and now adding mine and getting it integrated into docker.

**Pull request merged**